src/d3a/models/base.py

```diff
@@ -31,8 +32,17 @@
     from d3a.models.state import StateInterface
 
 
+class AssetType(Enum):
+    PRODUCER = 0
+    CONSUMER = 1
+    PROSUMER = 2
+    UNDEFINED = 3
+
+
 class AreaBehaviorBase:
     """Base class used by area behaviour defining classes `BaseStrategy`"""
+    asset_type: AssetType = AssetType.UNDEFINED
```

Comment:
This contradicts with the property you implemented below, right?
IMO this can/should be removed here.

Reply:
Indeed, removing this. Thanks!